diff --git a/node_modules/eleventy-plugin-slide-decks/slide-deck.webc b/node_modules/eleventy-plugin-slide-decks/slide-deck.webc
index 4d10a5f..4669e0c 100644
--- a/node_modules/eleventy-plugin-slide-decks/slide-deck.webc
+++ b/node_modules/eleventy-plugin-slide-decks/slide-deck.webc
@@ -100,6 +100,7 @@
     <main>
       <slidem-deck id="deck" @attributes>
         <sl-progress-bar slot="progress" id="slides-progress" indeterminate></sl-progress-bar>
+        <slot name="slides"></slot>
         <script webc:type="js"
                 webc:for="(slide, i) of getSlides($data, tag)">
           const tag = slide.data.is || 'slidem-slide';
